scripts/safe_template_linter.py

```diff
@@ -1254,6 +1272,202 @@ def _find_unescaped_expressions(self, underscore_template):
+        """
+        Process file to determine if it is a JavaScript file and
+        if it is safe.
+
+        Arguments:
+            directory (string): The directory of the file to be checked
+            file_name (string): A filename for a potential JavaScript file
+
+        Returns:
+            The file results containing any violations.
+
+        """
+        file_full_path = os.path.normpath(directory + '/' + file_name)
+        results = FileResults(file_full_path)
+
+        if not results.is_file:
+            return results
+
+        if not self._is_valid_directory(directory):
+            return results
+
+        if not file_name.lower().endswith('.js') or file_name.lower().endswith('.min.js'):
+            return results
+
+        return self._load_and_check_javascript_file_is_safe(file_full_path, results)
+
+    def _is_valid_directory(self, directory):
+        """
+        Determines if the provided directory is a directory that could contain
+        a JavaScript file that needs to be linted.
+
+        Arguments:
+            directory: The directory to be linted.
+
+        Returns:
+            True if this directory should be linted for JavaScript violations
+            and False otherwise.
+        """
+        if _is_skip_dir(self._skip_javascript_dirs, directory):
+            return False
+
+        return True
+
+    def _load_and_check_javascript_file_is_safe(self, file_full_path, results):
+        """
+        Loads the JavaScript file and checks if it is in violation.
+
+        Arguments:
+            file_full_path: The file to be loaded and linted.
+
+        Returns:
+            The file results containing any violations.
+
+        """
+        file_contents = _load_file(self, file_full_path)
+        self._check_javascript_file_is_safe(file_contents, results)
+        return results
+
+    def _check_javascript_file_is_safe(self, file_contents, results):
+        """
+        Checks for violations in a JavaScript file.
+
+        Arguments:
+            file_contents: The contents of the JavaScript file.
+            results: A file results objects to which violations will be added.
+
+        """
+        self._check_jquery_append(file_contents, results)
+        self._check_concat_with_html(file_contents, results)
+        results.prepare_results(file_contents)
+
+    def _check_jquery_append(self, file_contents, results):
+        """
+        Checks that append() calls are acceptable.
+
+        Acceptable calls to append():
+        - HtmlUtils.append()
+        - the argument can end with ".el", ".$el" (with no concatenation)
+        - the argument can be a single variable ending in "El" or staring with
+            "$". For example, "testEl" or "$test".
+        - the argument can be a single string literal with no HTML tags
+        - the argument can start with "[" or "{". If so, we assume this is an
+            array or object append() call.
```

Comment:
Hi @andy-armstrong.  I'm trying to not have the linter complain in places where it doesn't need to complain. 

For the the rule above, I now realize that JavaScript objects and arrays don't have an append() function.  However, I'm not quite sure what is going on in some of these lines:
https://github.com/edx/edx-platform/blob/master/lms/templates/courseware/progress_graph.js#L50-51
https://github.com/edx/edx-platform/blob/master/lms/templates/courseware/progress_graph.js#L83-L87
Are these still all just JQuery append?  Some of these look like plain old arrays, so I was confused.  Should I just drop this exception altogether?